File: server/API7D/objet/Coordonnees.cs
using System.Text.Json.Serialization;

/// <summary>
/// Classe représentant les coordonnées X et Y.
/// </summary>
public class Coordonnees
{
    
    private int x;
    private int y;

    /// <summary>
    /// obtient ou définit la coordonnée X
    /// </summary>
    public int X 
    {
        get { return x; } 
        set { x = value; }
    }

    /// <summary>
    /// obtient ou définit la coordonnée Y
    /// </summary>
    public int Y 
    { 
        get { return y;}
        set { y = value;}
    }

    /// <summary>
    /// Initialise une nouvelle instance de la classe Coordonnees
    /// </summary>
    /// <param name="x">La coordonnées X</param>
    /// <param name="y">La coordonnées Y</param>
    public Coordonnees(int x, int y)
    {
        this.X = x;
        this.Y = y;
    }

    public override bool Equals(object obj)
    {
        bool result = false;
        if (obj is Coordonnees other)
        {

            result = this.X == other.X && this.Y == other.Y;
        }
        return result;
    }
}

File: app/src/main/java/com/example/spotthedifference/data/model/Coordonnees.java
package com.example.spotthedifference.data.model;

/**
 * Classe représentant les coordonnées X et Y.
 */
public class Coordonnees {
    private int x;
    private int y;

    /**
     * Constructeur avec paramètres pour initialiser les coordonnées.
     *
     * @param x La coordonnée X.
     * @param y La coordonnée Y.
     */
    public Coordonnees(int x, int y) {
        this.x = x;
        this.y = y;
    }

    /**
     * Récupère la valeur de la coordonnée X.
     *
     * @return La coordonnée X.
     */
    public int getX() {
        return x;
    }

    /**
     * Récupère la valeur de la coordonnée Y.
     *
     * @return La coordonnée Y.
     */
    public int getY() {
        return y;
    }

    /**
     * Définit la valeur de la coordonnée X.
     *
     * @param x La nouvelle coordonnée X.
     */
    public void setX(int x) {
        this.x = x;
    }

    /**
     * Définit la valeur de la coordonnée Y.
     *
     * @param y La nouvelle coordonnée Y.
     */
    public void setY(int y) {
        this.y = y;
    }
}
